shared/middleware/files_api.rb

```diff
@@ -92,6 +79,10 @@ def allowed_file_type?(endpoint, extension)
     # read the entire request before considering rejecting it, otherwise varnish
     # may return a 503 instead of whatever status code we specify.
     body = request.body.read
+
+    owner_id, _ = storage_decrypt_channel_id(encrypted_channel_id)
+    not_authorized unless owner_id == storage_id('user')
```

Comment:
In [`StorageApps`](https://github.com/code-dot-org/code-dot-org/blob/a97afe27784e18c2a5de6744945e00bd8ed95e13/shared/middleware/helpers/storage_apps.rb#L10) we set up `@storage_id` in the initializer.  Any reason not to do the same here?

(I agree that this class should own the `storage_id` rather than having it passed it.)